### PR TITLE
update make file and testing

### PR DESCRIPTION
**Pull request by h-bragg**
Update the Make command and travis to be a bit more modern and test more environments in travis.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,40 +1,51 @@
...
+
+.PHONY: install update update-lowest help
+.PHONY: test test-unit test-matrix test-matrix-lowest
 
 .SILENT: help
 
-install: ## Download the depenedencies then build the image :rocket:.
-	make 'composer-install --optimize-autoloader --ignore-platform-reqs'
-	docker build --tag ${PROJECT}:latest .
+install: ## Download the dependencies then build the image :rocket:.
```
Comment (biggianteye) [inline Makefile]: In `make help`, this shows up as this:

```
install                 Download the dependencies then build the image              rocket  .
```

(Scroll across to see the problem.)

Given this is intended for command line display, I don't think the `:rocket:` is useful.

File: Makefile [hunk 2]
```diff
@@ -1,40 +1,51 @@
...
 composer-%: ## Run a composer command, `make "composer-<command> [...]"`.
 	docker run -it --rm \
-	-v $$(pwd):/app \
-	-v ~/.composer:/root/composer \
-	-v ~/.ssh:/root/.ssh:ro \
-	composer/composer --ansi --no-interaction $*
+        -v $$(pwd):/app \
+        -v ~/.composer:/tmp \
+        -v ~/.ssh:/root/.ssh:ro \
+        composer --ansi --no-interaction $*
```
Comment (biggianteye) [inline Makefile:25]: Consider changing `$*` to `"$@"` to take into account spaces and handling quotes sentences as single values.

File: Makefile [hunk 3]
```diff
@@ -1,40 +1,51 @@
...
-	docker run --rm -it -v $$(pwd):/opt/${PROJECT} ${PROJECT} \
-	composer test --ansi
+	${DOCKER_RUN} vendor/bin/phpunit tests/
 
-test-matrix:
-	docker run --rm -it -v $$(pwd):/opt/${PROJECT} -w /opt/${PROJECT} php:5.6-cli \
-	vendor/bin/phpunit --testsuite unit
-	docker run --rm -it -v $$(pwd):/opt/${PROJECT} -w /opt/${PROJECT} php:7.0-cli \
-	vendor/bin/phpunit --testsuite unit
+test-matrix: ## Test in multiple images
```
Comment (biggianteye) [inline Makefile]: This implies using the latest compatible versions (as opposed to `test-matrix-lowest`). If you run `make test-matrix-lowest test-matrix` you will get misleading results. I think this should do an `update` first.

File: Makefile (hunk 2 above)
Comment (h-bragg) [reply Makefile:25]: `$*` has a sort of special meaning here (as in what was the `%` bit in the command):

so `make 'composer-install -o'` will produce `composer --ansi --no-interaction install -o` whereas adding `"$*"` or `"$@"` puts quotes around it so it calls `composer --ansi --no-interaction "install -o"`

File: Makefile [hunk 4]
```diff
@@ -1,40 +1,51 @@
...
+	${MAKE} test-unit IMAGE=php:5.6-alpine
+	${MAKE} test-unit IMAGE=php:7.0-alpine
+	${MAKE} test-unit IMAGE=php:7.1-alpine
 
-clean: ## Clean up any images.
-	docker rmi ${PROJECT}:latest
+test-matrix-lowest: ## Test multiple images
```
Comment (biggianteye) [inline Makefile]: The `clean` target has gone. I think this would still be useful to have for things like the vendor directory.

**Review by biggianteye — CHANGES_REQUESTED** [review]
A couple of minor changes listed above.

File: Makefile (hunk 2 above)
Comment (biggianteye) [reply Makefile:25]: Fair enough. I forgot the context. This isn't the shell!

File: Makefile (hunk 1 above)
Comment (h-bragg) [reply Makefile]: used a 'real' rocket: 🚀

File: Makefile (hunk 4 above)
Comment (h-bragg) [reply Makefile]: re-added clean target

File: Makefile (hunk 3 above)
Comment (h-bragg) [reply Makefile]: added update to test-matrix

**Review by biggianteye — APPROVED** [review]

**Pull request merged**